From bf8379ede7ba86d2853037e651022f184bf7af4e Mon Sep 17 00:00:00 2001
From: _ <>
Date: Tue, 26 May 2026 00:00:00 +0000
Subject: [PATCH] increase avx512 gather/scatter costs

---
 llvm/lib/Target/X86/X86TargetTransformInfo.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
index e3d838a7f..3ea654ed2 100644
--- a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
+++ b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
@@ -5110,7 +5110,7 @@ InstructionCost X86TTIImpl::getVectorInstrCost(
 
     // Consider cheap cases.
     if (IsCheapPInsrPExtrInsertPS())
-      return 1 + RegisterFileMoveCost;
+      return 100 + RegisterFileMoveCost;
 
     // For extractions we just need to shuffle the element to index 0, which
     // should be very cheap (assume cost = 1). For insertions we need to shuffle
@@ -6596,14 +6596,14 @@ int X86TTIImpl::getGatherOverhead() const {
   // TODO: Remove the explicit hasAVX512()?, That would mean we would only
   // enable gather with a -march.
   if (ST->hasAVX512() || (ST->hasAVX2() && ST->hasFastGather()))
-    return 2;
+    return 10;
 
   return 1024;
 }
 
 int X86TTIImpl::getScatterOverhead() const {
   if (ST->hasAVX512())
-    return 2;
+    return 10;
 
   return 1024;
 }
-- 
2.55.0

