From 41419fa94ed7e5c897b36a8f632f76a88a392171 Mon Sep 17 00:00:00 2001
From: _ <>
Date: Tue, 26 May 2026 00:00:00 +0000
Subject: [PATCH] increase avx512 gather/scatter costs

---
 llvm/lib/Target/X86/X86TargetTransformInfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
index e80fd692df..b97911b7a6 100644
--- a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
+++ b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
@@ -6435,14 +6435,14 @@ int X86TTIImpl::getGatherOverhead() const {
   // TODO: Remove the explicit hasAVX512()?, That would mean we would only
   // enable gather with a -march.
   if (ST->hasAVX512() || (ST->hasAVX2() && ST->hasFastGather()))
-    return 2;
+    return 16;
 
   return 1024;
 }
 
 int X86TTIImpl::getScatterOverhead() const {
   if (ST->hasAVX512())
-    return 2;
+    return 8;
 
   return 1024;
 }
-- 
2.55.0

