From 7e726278261be54bcd03905db806144eb076d9a4 Mon Sep 17 00:00:00 2001
From: Simon Pilgrim <llvm-dev@redking.me.uk>
Date: Thu, 27 Nov 2025 15:03:10 +0000
Subject: [PATCH] add avx512 MULHS/MULHU lowering

---
 .../lib/Target/X86/X86TargetTransformInfo.cpp | 42 +++++++++++++++----
 1 file changed, 34 insertions(+), 8 deletions(-)

diff --git a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
index e629bbb60..af9eba09e 100644
--- a/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
+++ b/llvm/lib/Target/X86/X86TargetTransformInfo.cpp
@@ -450,6 +450,16 @@ InstructionCost X86TTIImpl::getArithmeticInstrCost(
     { ISD::SREM, MVT::v16i32, {  8 } }, // pmuludq+mul+sub sequence
     { ISD::UDIV, MVT::v16i32, {  5 } }, // pmuludq sequence
     { ISD::UREM, MVT::v16i32, {  7 } }, // pmuludq+mul+sub sequence
+
+    { ISD::SDIV, MVT::v8i64,  {  6 } },
+    { ISD::SREM, MVT::v8i64,  {  8 } },
+    { ISD::UDIV, MVT::v8i64,  {  5 } },
+    { ISD::UREM, MVT::v8i64,  {  7 } },
+
+    { ISD::SDIV, MVT::v4i64,  {  6 } },
+    { ISD::SREM, MVT::v4i64,  {  8 } },
+    { ISD::UDIV, MVT::v4i64,  {  5 } },
+    { ISD::UREM, MVT::v4i64,  {  7 } },
   };
 
   if (Op2Info.isUniform() && Op2Info.isConstant() && ST->hasAVX512())
@@ -491,6 +501,9 @@ InstructionCost X86TTIImpl::getArithmeticInstrCost(
     { ISD::SREM, MVT::v8i32, {  8 } }, // pmuludq+mul+sub sequence
     { ISD::UDIV, MVT::v8i32, {  5 } }, // pmuludq sequence
     { ISD::UREM, MVT::v8i32, {  7 } }, // pmuludq+mul+sub sequence
+
+    { ISD::UDIV, MVT::v4i64, {  5 } },
+    { ISD::UREM, MVT::v4i64, {  7 } },
   };
 
   if (Op2Info.isUniform() && Op2Info.isConstant() && ST->hasAVX2())
@@ -602,10 +615,20 @@ InstructionCost X86TTIImpl::getArithmeticInstrCost(
     { ISD::UDIV, MVT::v32i16, { 12 } }, // 2*vpmulhuw sequence
     { ISD::UREM, MVT::v32i16, { 16 } }, // 2*vpmulhuw+mul+sub sequence
 
-    { ISD::SDIV, MVT::v16i32, { 15 } }, // vpmuldq sequence
-    { ISD::SREM, MVT::v16i32, { 17 } }, // vpmuldq+mul+sub sequence
-    { ISD::UDIV, MVT::v16i32, { 15 } }, // vpmuludq sequence
-    { ISD::UREM, MVT::v16i32, { 17 } }, // vpmuludq+mul+sub sequence
+    { ISD::SDIV, MVT::v16i32, {  6 } }, // vpmuldq sequence
+    { ISD::SREM, MVT::v16i32, {  8 } }, // vpmuldq+mul+sub sequence
+    { ISD::UDIV, MVT::v16i32, {  5 } }, // vpmuludq sequence
+    { ISD::UREM, MVT::v16i32, {  7 } }, // vpmuludq+mul+sub sequence
+
+    { ISD::SDIV, MVT::v8i64,  {  6 } },
+    { ISD::SREM, MVT::v8i64,  {  8 } },
+    { ISD::UDIV, MVT::v8i64,  {  5 } },
+    { ISD::UREM, MVT::v8i64,  {  7 } },
+
+    { ISD::SDIV, MVT::v4i64,  {  6 } },
+    { ISD::SREM, MVT::v4i64,  {  8 } },
+    { ISD::UDIV, MVT::v4i64,  {  5 } },
+    { ISD::UREM, MVT::v4i64,  {  7 } },
   };
 
   if (Op2Info.isConstant() && ST->hasAVX512())
@@ -625,10 +648,13 @@ InstructionCost X86TTIImpl::getArithmeticInstrCost(
     { ISD::UDIV, MVT::v16i16, {  6 } }, // vpmulhuw sequence
     { ISD::UREM, MVT::v16i16, {  8 } }, // vpmulhuw+mul+sub sequence
 
-    { ISD::SDIV, MVT::v8i32,  { 15 } }, // vpmuldq sequence
-    { ISD::SREM, MVT::v8i32,  { 19 } }, // vpmuldq+mul+sub sequence
-    { ISD::UDIV, MVT::v8i32,  { 15 } }, // vpmuludq sequence
-    { ISD::UREM, MVT::v8i32,  { 19 } }, // vpmuludq+mul+sub sequence
+    { ISD::SDIV, MVT::v8i32,  {  6 } }, // vpmuldq sequence
+    { ISD::SREM, MVT::v8i32,  {  8 } }, // vpmuldq+mul+sub sequence
+    { ISD::UDIV, MVT::v8i32,  {  5 } }, // vpmuludq sequence
+    { ISD::UREM, MVT::v8i32,  {  7 } }, // vpmuludq+mul+sub sequence
+
+    { ISD::UDIV, MVT::v4i64,  {  5 } },
+    { ISD::UREM, MVT::v4i64,  {  7 } },
   };
 
   if (Op2Info.isConstant() && ST->hasAVX2())
-- 
2.55.0

