From b5fa759a7cd810676c22670480663b7b7b17b2fa Mon Sep 17 00:00:00 2001
From: _ <>
Date: Fri, 8 Aug 2025 00:00:00 +0000
Subject: [PATCH] fuck curl

---
 src/tool_cfgable.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tool_cfgable.c b/src/tool_cfgable.c
index 015ee26..11357d8 100644
--- a/src/tool_cfgable.c
+++ b/src/tool_cfgable.c
@@ -40,6 +40,8 @@ struct OperationConfig *config_alloc(void)
   if(!config)
     return NULL;
 
+  config->proxy_native_ca_store = TRUE;
+  config->native_ca_store = TRUE;
   config->use_httpget = FALSE;
   config->create_dirs = FALSE;
   config->maxredirs = DEFAULT_MAXREDIRS;
-- 
2.50.1

