From a8264ee2f9e25f3907627baeed95b2caeeaff23b Mon Sep 17 00:00:00 2001
From: _ <>
Date: Wed, 10 Dec 2025 00:00:00 +0000
Subject: [PATCH] libass: use avx2 unconditionally if defined avx2

---
 libass/ass_bitmap_engine.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/libass/ass_bitmap_engine.c b/libass/ass_bitmap_engine.c
index 847adc4..4e926e3 100644
--- a/libass/ass_bitmap_engine.c
+++ b/libass/ass_bitmap_engine.c
@@ -168,7 +168,14 @@ BitmapEngine ass_bitmap_engine_init(unsigned mask)
     engine.tile_order = mask & ASS_FLAG_LARGE_TILES ? 5 : 4;
 
 #if CONFIG_ASM
-    unsigned flags = ass_get_cpu_flags(mask);
+    unsigned flags =
+#if ARCH_X86 && defined(__AVX2__)
+    ASS_CPU_FLAG_X86_AVX2;
+#elif ARCH_AARCH64 && defined(__ARM_NEON)
+    ASS_CPU_FLAG_ARM_NEON;
+#else
+    ass_get_cpu_flags(mask);
+#endif
 #if ARCH_X86
     if (flags & ASS_CPU_FLAG_X86_AVX2) {
         ALL_PROTOTYPES(32, avx2)
-- 
2.52.0

