From ff8879a0e07dcc4fc5eb9bbc8790cfa7e8da7352 Mon Sep 17 00:00:00 2001
From: _ <>
Date: Thu, 22 Jan 2026 00:00:00 +0000
Subject: [PATCH] add static dispatch for aarch64

---
 crypto/armcap.c | 31 ++++++++++++++++++++++++++++---
 1 file changed, 28 insertions(+), 3 deletions(-)

diff --git a/crypto/armcap.c b/crypto/armcap.c
index d8ad9ce01..2095f72ff 100644
--- a/crypto/armcap.c
+++ b/crypto/armcap.c
@@ -41,13 +41,38 @@ void OPENSSL_cpuid_setup(void)
 {
     OPENSSL_armcap_P |= ARMV7_NEON;
     OPENSSL_armv8_rsa_neonized = 1;
-    if (IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE)) {
-        /* These are all covered by one call in Windows */
+    #if defined(__ARM_FEATURE_CRYPTO)
         OPENSSL_armcap_P |= ARMV8_AES;
         OPENSSL_armcap_P |= ARMV8_PMULL;
         OPENSSL_armcap_P |= ARMV8_SHA1;
         OPENSSL_armcap_P |= ARMV8_SHA256;
-    }
+    #else
+        if (IsProcessorFeaturePresent(PF_ARM_V8_CRYPTO_INSTRUCTIONS_AVAILABLE)) {
+            /* These are all covered by one call in Windows */
+            OPENSSL_armcap_P |= ARMV8_AES;
+            OPENSSL_armcap_P |= ARMV8_PMULL;
+            OPENSSL_armcap_P |= ARMV8_SHA1;
+            OPENSSL_armcap_P |= ARMV8_SHA256;
+        }
+    #endif
+    #if defined(__ARM_FEATURE_SHA512)
+        OPENSSL_armcap_P |= ARMV8_SHA512;
+    #endif
+    #if defined(__ARM_FEATURE_SHA3)
+        OPENSSL_armcap_P |= ARMV8_SHA3;
+    #endif
+    #if defined(__ARM_FEATURE_SM3)
+        OPENSSL_armcap_P |= ARMV8_SM3;
+    #endif
+    #if defined(__ARM_FEATURE_SM4)
+        OPENSSL_armcap_P |= ARMV8_SM4;
+    #endif
+    #if defined(__ARM_FEATURE_SVE)
+        OPENSSL_armcap_P |= ARMV8_SVE;
+    #endif
+    #if defined(__ARM_FEATURE_SVE2)
+        OPENSSL_armcap_P |= ARMV9_SVE2;
+    #endif
 }
 
 uint32_t OPENSSL_rdtsc(void)
-- 
2.53.0

