From 17e85dd4790d4991ed986bc65d22288747249d4e Mon Sep 17 00:00:00 2001
From: _ <>
Date: Thu, 22 Jan 2026 00:00:00 +0000
Subject: [PATCH] add static dispatch for aarch64

---
 crypto/armcap.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/crypto/armcap.c b/crypto/armcap.c
index d8ad9ce..efc0594 100644
--- a/crypto/armcap.c
+++ b/crypto/armcap.c
@@ -48,6 +48,30 @@ void OPENSSL_cpuid_setup(void)
         OPENSSL_armcap_P |= ARMV8_SHA1;
         OPENSSL_armcap_P |= ARMV8_SHA256;
     }
+    #if defined(__ARM_FEATURE_CRYPTO)
+        OPENSSL_armcap_P |= ARMV8_AES;
+        OPENSSL_armcap_P |= ARMV8_PMULL;
+        OPENSSL_armcap_P |= ARMV8_SHA1;
+        OPENSSL_armcap_P |= ARMV8_SHA256;
+    #endif
+    #if defined(__ARM_FEATURE_SHA512)
+        OPENSSL_armcap_P |= ARMV8_SHA512;
+    #endif
+    #if defined(__ARM_FEATURE_SHA3)
+        OPENSSL_armcap_P |= ARMV8_SHA3;
+    #endif
+    #if defined(__ARM_FEATURE_SM3)
+        OPENSSL_armcap_P |= ARMV8_SM3;
+    #endif
+    #if defined(__ARM_FEATURE_SM4)
+        OPENSSL_armcap_P |= ARMV8_SM4;
+    #endif
+    #if defined(__ARM_FEATURE_SVE)
+        OPENSSL_armcap_P |= ARMV8_SVE;
+    #endif
+    #if defined(__ARM_FEATURE_SVE2)
+        OPENSSL_armcap_P |= ARMV9_SVE2;
+    #endif
 }
 
 uint32_t OPENSSL_rdtsc(void)
-- 
2.52.0

