From fb0aa132b9ae2bbd0a13ce6e79a1ffeb0dd53615 Mon Sep 17 00:00:00 2001
From: _ <>
Date: Sat, 17 May 2025 00:00:00 +0000
Subject: [PATCH] update manifest

---
 src/qbittorrent.exe.manifest | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/qbittorrent.exe.manifest b/src/qbittorrent.exe.manifest
index 9094098..7b87007 100644
--- a/src/qbittorrent.exe.manifest
+++ b/src/qbittorrent.exe.manifest
@@ -42,6 +42,11 @@
       <heapType xmlns="http://schemas.microsoft.com/SMI/2020/WindowsSettings">SegmentHeap</heapType>
       <!-- Enable long paths that exceed MAX_PATH in length -->
       <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2,PerMonitor</dpiAwareness>
+      <printerDriverIsolation xmlns="http://schemas.microsoft.com/SMI/2011/WindowsSettings">True</printerDriverIsolation>
+      <disableWindowFiltering xmlns="http://schemas.microsoft.com/SMI/2011/WindowsSettings">True</disableWindowFiltering>
+      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">True/PM</dpiAware>
     </windowsSettings>
   </application>
 </assembly>
-- 
2.49.0

